Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2AckermannModelHook.cpp

```diff
@@ -0,0 +1,320 @@
+{
+    namespace AckermannParser
+    {
+        static constexpr float epsilon = 0.001f;
+
+        namespace Wheels
+        {
+            static constexpr unsigned int FrontLeft = 0;
+            static constexpr unsigned int FrontRight = 1;
+            static constexpr unsigned int RearLeft = 2;
+            static constexpr unsigned int RearRight = 3;
+            static constexpr unsigned int SteeringLeft = 4;
+            static constexpr unsigned int SteeringRight = 5;
+            static constexpr unsigned int Total = 6;
+        } // namespace Wheels
+
+        AZ::Outcome<AZ::Vector3, AZStd::string> GetPosition(const AZ::Entity* entity)
+        {
+            if (entity)
+            {
+                auto* transformInterface = entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+                if (transformInterface)
+                {
+                    return AZ::Success(transformInterface->GetWorldTM().GetTranslation());
+                }
+            }
+
+            return AZ::Failure(AZStd::string("Failed to obtain position parameter for wheel entity."));
+        }
+
+        float GetTrack(const AZ::Entity* entityLeft, const AZ::Entity* entityRight)
+        {
+            const auto positionLeft = GetPosition(entityLeft);
+            const auto positionRight = GetPosition(entityRight);
+
+            if (positionLeft.IsSuccess() && positionLeft.IsSuccess())
+            {
+                return positionLeft.GetValue().GetDistance(positionRight.GetValue());
+            }
+
+            return 0.0f;
+        }
+
+        bool IsSteeringWheel(const std::string& jointNameSteering, const std::string& jointNameWheel, const sdf::Model& sdfModel)
+        {
+            const auto jointPtrSteering = sdfModel.JointByName(jointNameSteering);
+            const auto jointPtrWheel = sdfModel.JointByName(jointNameWheel);
+            if (jointPtrSteering != nullptr && jointPtrWheel != nullptr)
+            {
+                return (jointPtrSteering->ChildName() == jointPtrWheel->ParentName());
+            }
+
+            return false;
+        }
+
+        float GetWheelRadius(AZ::Entity* entity)
+        {
+            if (entity != nullptr)
+            {
+                PhysX::EditorColliderComponent* colliderComponent = entity->FindComponent<PhysX::EditorColliderComponent>();
+                if (colliderComponent != nullptr)
+                {
+                    float radius = 0.0f;
+                    entity->Activate();
+                    if (entity->GetState() == AZ::Entity::State::Active)
+                    {
+                        PhysX::EditorPrimitiveColliderComponentRequestBus::EventResult(
+                            radius,
+                            AZ::EntityComponentIdPair(entity->GetId(), colliderComponent->GetId()),
+                            &PhysX::EditorPrimitiveColliderComponentRequests::GetCylinderRadius);
+                        entity->Deactivate();
```

Comment:
I added an assert. The assumption is it would only work in the editor, hence the entity should not be active beforehand. It needs to be activated to get the radius.